src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -529,115 +546,20 @@ public DataAccessFactory getDataStoreFactory(DataStoreInfo info) throws IOExcept
 
         DataAccess<? extends FeatureType, ? extends Feature> dataStore = null;
         try {
-            String id = info.getId();
-            dataStore = dataStoreCache.get(id);
+
+            String storeId = info.getId();
+            // cache only if the id is not null, no need to cache the stores
```

Comment:
Netiher the previous code was putting the store in the cache. For what I saw debugging, a store with null id can comes only when saving a new store [here ](https://github.com/geoserver/geoserver/blob/71bfcd03119c2bba41e3c8133b6c520e62f691ac/src/web/core/src/main/java/org/geoserver/web/data/store/DataAccessNewPage.java#L91), because the id is set later from the catalog, and the data access get then disposed correctly.